src/schema/city/index.ts

```diff
@@ -100,13 +104,29 @@ export const City = {
   description: "A city-based entry point for local discovery",
   args: {
     slug: {
-      type: new GraphQLNonNull(GraphQLString),
+      type: GraphQLString,
       description:
         "A slug for the city, conforming to Gravity's city slug naming conventions",
     },
+    near: {
+      type: Near,
+      description:
+        "A point which will be used to locate the nearest local discovery city within a threshold",
+    },
   },
-  resolve: (_obj, args) => {
-    return lookupCity(args.slug)
+  resolve: (_obj, { slug, near }) => {
+    if (slug && near) {
+      throw new Error('The "slug" and "near" arguments are mutually exclusive.')
+    }
+    if (!slug && !near) {
+      throw new Error('One of the arguments "slug" or "near" is required.')
+    }
```

Comment:
Now that we allow `slug` XOR `near` as arguments to the query, I do some manual arg-checking here, as my understanding is that there is no native GraphQL notion of mutually exclusive arguments.

Reply:
Alas not.